File: MinJu/240108_week/Programmers_12899_124 나라의 숫자.py
# 124 나라의 숫자

# 자연수만 존재 + 1, 2, 4로 표현
def solution(n):
    answer = ''
    # 1, 4, 7, 10 ,,,, 은 일의 자리 1
    # 3의 배수 제외 3진법
    # 3의 배수이면 n//3 -1
    while n:
        if n % 3 != 0: # 3의 배수가 아니라면
            answer += str(n%3)
            n = n//3
        else:
            answer += "4"
            n = n//3 -1
    return answer[::-1] # 역순

File: SooYeon/240227_week/Programmers_12899_124나라의 숫자.js
function solution(n) {
  let answer = '';
  let value = [4, 1, 2];
  
  while(n) {
      answer = value[n % 3] + answer;
      n = n % 3 === 0 ? Math.floor((n - 1) / 3) : Math.floor(n / 3);
  }
  
  return answer;
}